Shared/Samples/Show device location history/ShowDeviceLocationHistoryView.swift

```diff
@@ -0,0 +1,234 @@
+        
+        init() {
+            locationDisplay = LocationDisplay(dataSource: Self.makeSimulatedLocationDataSource())
+            locationDisplay.autoPanMode = .recenter
+            
+            trackOverlay = GraphicsOverlay(graphics: [trackGraphic])
+            trackOverlay.renderer = SimpleRenderer(
+                symbol: SimpleLineSymbol(
+                    style: .solid,
+                    color: UIColor(Color.accentColor.opacity(0.5)),
+                    width: 3
+                )
+            )
+            
+            locationsOverlay = GraphicsOverlay(graphics: [locationGraphic])
+            locationsOverlay.renderer = SimpleRenderer(
+                symbol: SimpleMarkerSymbol(
+                    style: .circle,
+                    color: .red,
+                    size: 5
+                )
+            )
+        }
+        
+        /// Starts the location data source.
+        func startLocationDataSource() async throws {
+            guard locationDisplay.dataSource.status != .started else { return }
+            try await locationDisplay.dataSource.start()
+        }
+        
+        /// Stops the location data source.
+        func stopLocationDataSource() {
+            Task { [weak self] in
+                await self?.locationDisplay.dataSource.stop()
+            }
+        }
+        
+        /// A helper method to visualize the traveled path.
+        /// - Parameter lastPosition: The last location point from the
+        /// location data source.
+        private func processLocationUpdate(lastPosition: Point) {
+            pointBuilder.points.append(lastPosition)
+            locationGraphic.geometry = pointBuilder.toGeometry()
+            trackBuilder.add(lastPosition)
+            trackGraphic.geometry = trackBuilder.toGeometry()
+        }
+        
+        /// Starts to process location updates produced by the async stream
+        /// using the `for-await-in` syntax.
+        func startProcessingLocationChanges() {
+            locationProcessingTask = Task { [weak self] in
+                for await newLocation in locationDisplay.dataSource.locations {
+                    self?.processLocationUpdate(lastPosition: newLocation.position)
+                }
+            }
+        }
+        
+        /// Stops processing location updates.
+        func stopProcessingLocationChanges() {
+            locationProcessingTask?.cancel()
+            locationProcessingTask = nil
+        }
+        
+        /// Makes the simulated location data source for the sample.
+        /// - Returns: A simulated location data source in Los Angeles, CA.
+        private static func makeSimulatedLocationDataSource() -> SimulatedLocationDataSource {
+            let routePolyline = try? Polyline.fromJSON(polylineJSONString)
+            // Densify the simulated path to make it smoother.
+            let densifiedRoute = GeometryEngine.geodeticDensify(
+                routePolyline!,
```

Comment:
```suggestion
            let routePolyline = try! Polyline.fromJSON(polylineJSONString)
            // Densify the simulated path to make it smoother.
            let densifiedRoute = GeometryEngine.geodeticDensify(
                routePolyline,
```

Just noticed that this generates a linting error, so this suggestion shouldn't be accepted. However, the suggested code makes more sense that conditionally trying something and just forcing it later, so perhaps we should reconsider having the `force_try` rule enabled.

Reply:
I don't want to leave `force_try` out - in most cases it is code smell. This is some of the rare cases that we actually need to `try!`, so instead I'll add a line to disable it. // swiftlint:disable:next force_try